app/models/ability.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
+    if user.admin?
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, Post
+      can :create, Comment
+      can :create, Like
+
+      can :destroy, Post do |post|
+        post.author_id == user.id
+      end
+      can :destroy, Comment do |comment|
+        comment.author_id == user.id
+      end
```

Comment:
- [ ] There is no need for write seperate for all create , destroy , you can directy use :- 
 ```ruby
can :manage, Post, author: user         # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
```